app/bundles/FormBundle/Tests/Entity/FieldTest.php

```diff
@@ -328,4 +329,65 @@ public static function dataProvider(): iterable
         yield ['checkboxgrp', ['multiple' => 0], true];
         yield ['checkboxgrp', ['multiple' => 1], true];
     }
+
+    public function testCheckboxGroupOptionsWithCustomLeadFieldInForm(): void
+    {
+        // Create a custom boolean field for contacts.
+        $customField = new LeadField();
+        $customField->setObject('lead');
+        $customField->setType('boolean');
+        $customField->setLabel('Test Custom Field');
+        $customField->setAlias('custom_boolean_field');
+        $customField->setProperties([
+            'yes' => 'Yes',
+            'no'  => 'No',
+        ]);
+
+        // Create a new form
+        $form = new Form();
+        $form->setName('Test Form');
+        $form->setAlias('test_form');
+
+        // Create a checkbox group field
+        $field = new Field();
+        $field->setType('checkboxgrp');
+        $field->setLabel('Test Checkbox Group');
+        $field->setAlias('test_checkbox_group');
+
+        // Map to custom field
+        $field->setMappedField('custom_boolean_field');
+        $field->setMappedObject('contact');
+
+        // Add properties
+        $fieldProperties = [
+            'list' => [
+                'option1' => 'First Option',
+                'option2' => 'Second Option',
+            ],
+        ];
+        $field->setProperties($fieldProperties);
+
+        // Add field to form
+        $form->addField(0, $field);
+        $field->setForm($form);
+
+        // Verify the options
```

Comment:
I know this is giving a false positive result. I want to check the rendering values, but I can't find a proper way to do it in the test. Please suggest the best approach I can take here.